packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -49,6 +49,14 @@ function addVNodeToChildLWC(vnode: VCustomElement) {
     ArrayPush.call(getVMBeingRendered()!.velements, vnode);
 }
 
+// [s]et [o]wner
+function so(vnode: VNode): VNode {
+    vnode.owner = getVMBeingRendered()!;
+    vnode.isStatic = true;
+
+    return vnode;
+}
```

Comment:
I don't think we can run in cases where the owner is incorrectly set: this function (`so`) is invoked in the template evaluation, therefore, it will always have the correct owner. Notice that nothing is done for `patch`.

This PR does not need #2716, however, it can benefit from it, because then there's no need to have this function.

Reply:
I'm not sure that's true... the owner might be used somewhere else... I think @pmdartus is correct here. This is what of those issues that I believe are fatal of trying to re-use an actual vnode.

Reply:
which is that case? to the best of my knowledge, the vnodes are kept for reference only to do the diffing.